core/execution/future/liquidation.go

```diff
@@ -0,0 +1,94 @@
+package future
+
+import (
+	"context"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/logging"
+)
+
+func (m *Market) checkNetwork(ctx context.Context, now time.Time) error {
+	// debug
+	// this only returns an error if we couldn't get the price range, incidating no orders on book
```

Comment:
Feels a bit fragile that we're relying on the code path all the way into the matching engine to always only return an error when there are no orders on the book. Can we not catch the specific error, or change `OnTick()` to return `nil, nil` in that case so at least _here_ we can handle errors properly?

Reply:
This is fine here, in part because this isn't yet the final form of the liquidation implementation. The liquidation engine will do more heavy lifting come the next PR. I agree that this has the whiff of fragile code, but it's not intended to survive the next PR